realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -426,6 +427,68 @@ public SyncUser run() {
         }.start();
     }
 
+    /**
+     * Helper method for Admin users in order to look up a {@code SyncUser} using the identity provider and the used username.
+     * @param provider identity providers {@link io.realm.SyncCredentials.IdentityProvider}.
+     * @param providerId username or email used to create the account first time.
+     * @return {@code SyncUser} associated with the given identity provider and providerId.
+     * @throws ObjectServerError if the user could not be found.
+     */
+    public SyncUser retrieveUser(final String provider, final String providerId) throws ObjectServerError {
+        if (Util.isEmptyString(provider)) {
+            throw new IllegalArgumentException("Not-null 'provider' required.");
+        }
+
+        if (Util.isEmptyString(providerId)) {
+            throw new IllegalArgumentException("None empty 'providerId' required.");
+        }
+
+        if (!isAdmin()) {
+            throw new IllegalStateException("User need to be admin in order to lookup ID.");
+        }
+
+        AuthenticationServer authServer = SyncManager.getAuthServer();
+        LookupUserIdResponse response = authServer.retrieveUser(getSyncUser().getUserToken(), provider, providerId, getAuthenticationUrl());
+        if (!response.isValid()) {
+            throw response.getError();
+        } else {
+            SyncUser syncUser = SyncManager.getUserStore().get(response.getUserId());
+            if (syncUser != null) {
+                return syncUser;
+            } else {
+                // build an SynUser without a token
+                Token refreshToken = new Token(null, response.getUserId(), null, 0, null, response.isAdmin());
+                ObjectServerUser objectServerUser = new ObjectServerUser(refreshToken, getAuthenticationUrl());
+                objectServerUser.localLogout();
+                return new SyncUser(objectServerUser);
+            }
+        }
+    }
+
+    /**
+     * Asynchronously lookup up a {@code SyncUser} using the identity provider and the used username.
+     * This is for Admin users only.
+     *
+     * @param provider identity providers {@link io.realm.SyncCredentials.IdentityProvider}.
```

Comment:
{@link io.realm.SyncCredentials.IdentityProvider} used when the account was created.